### ISSUE TITLE
any way to stop the provider?

### ISSUE THREAD

**Issue by wighawag** [ConsenSys-archive/ganache-cli-archive#70]
Currently testrpc provider is waiting/listenning for ever, is there any way to make it stop?

**Comment by tcoulter** [comment]
Hi @wighawag. I need more context here. Can you provide code showing how you are using it?

**Comment by wighawag** [comment]
content of test.js:

```
var TestRPC = require("ethereumjs-testrpc");
var provider = TestRPC.provider();
```

execute :

```
node test.js
```

there is no way to stop this except by quitting the app. It keep listenning I guess.
it would be nice if we could stop it like `provider.stop()`

I need this to be able to test full test scenario from fresh accounts

**Comment by tcoulter** [comment]
Hmm, as far as I understand, you shouldn't _have_ to call `provider.stop()` if it existed. I'd consider keeping the process open as a bug.

I'm guessing this has something to do with a timer somewhere. Let me investigate.

**Comment by tcoulter** [comment]
Definitely a bug. Thanks for reporting!

**Comment by kumavis** [comment]
> `provider.stop()`

https://github.com/MetaMask/provider-engine/blob/master/index.js#L39

however this won't be necessary after we make optional the block polling mechanism

**Comment by wighawag** [comment]
that's cool. thanks

**Comment by wighawag** [comment]
When can I expect the new version of web3-provider-engine be incorporated in testrpc / submited to npm ?

**Comment by kumavis** [comment]
after i've written it ; )
idk, two wooks™

**Comment by graup** [comment]
Just as a thought: For testing, I start testrpc via [concurrently](https://www.npmjs.com/package/concurrently) and have it exit automatically when the tests finish:

    concurrently --kill-others -s first "testrpc" "npm run test"

**Comment by tcoulter** [comment]
[This pull request](https://github.com/MetaMask/provider-engine/pull/81) should fix your issue once it's been merged.

**Comment by tcoulter** [comment]
This has been fixed, and is now available in v2.2.4. You shouldn't have to stop the engine anymore, hence no need to expose it. Cheers!
